web3.js/src/epoch-schedule.ts

```diff
@@ -3,16 +3,29 @@ const MINIMUM_SLOT_PER_EPOCH = 32;
 // Returns the number of trailing zeros in the binary representation of self.
 function trailingZeros(n: number) {
   let trailingZeros = 0;
-  while ((n & 1) == 0) {
+  while (n > 1) {
     n /= 2;
     trailingZeros++;
   }
   return trailingZeros;
 }
 
+// Returns the smallest power of two greater than or equal to n
+function nextPowerOfTwo(n: number) {
+  if (n === 0) return 1;
+  n--;
+  n |= n >> 1;
+  n |= n >> 2;
+  n |= n >> 4;
+  n |= n >> 8;
+  n |= n >> 16;
```

Comment:
I think you need one more iteration because `n` can be greater than `2^32`

Reply:
correct, i blindly followed a solution i found and forgot we are "dealing" with u64 here